adserver/analyzer/models.py

```diff
@@ -68,5 +62,31 @@ def save(self, *args, **kwargs):
         self.full_clean()
         return super().save(*args, **kwargs)
 
+    class Meta:
+        abstract = True
+
+
+class AnalyzedUrl(BaseAnalyzedUrl):
+    """Analyzed keywords for a given URL."""
+
+    publisher = models.ForeignKey(
+        Publisher,
+        help_text=_("Publisher where this URL appears"),
+        on_delete=models.CASCADE,
+    )
+
     class Meta:
         unique_together = ("url", "publisher")
+
+
+class AnalyzedAd(BaseAnalyzedUrl):
```

Comment:
I sort of dislike this name. `AnalyzedAd` sounds like it FKs to advertisement which it doesn't. This should probably be named `AnalyzedAdvertiserUrl` or something similar.